plugins/inputs/http_response/http_response.go

```diff
@@ -138,14 +145,43 @@ func localAddress(interfaceName string) (net.Addr, error) {
 		return nil, err
 	}
 
+	urlInIPv6, zone := isURLInIPv6(address)
 	for _, addr := range addrs {
 		if naddr, ok := addr.(*net.IPNet); ok {
-			// leaving port set to zero to let kernel pick
-			return &net.TCPAddr{IP: naddr.IP}, nil
+			ipNetInIPv6 := isIPNetInIPv6(naddr)
+
+			// choose interface address in the same format as server address
+			if ipNetInIPv6 == urlInIPv6 {
+				// leaving port set to zero to let kernel pick, but set zone
+				return &net.TCPAddr{IP: naddr.IP, Zone: zone}, nil
+			}
+		}
+	}
+
+	return nil, fmt.Errorf("cannot create local address for interface %q and server address %q", interfaceName, address.String())
```

Comment:
This seems to break the case when the address is a hostname instead of an ip address:

```
Error running agent: could not initialize input inputs.http_response: cannot create local address for interface "tinc.retiolum" and server address "http://transmission.r/transmission/web/"
```